go/parquet/file/page_writer.go

```diff
@@ -451,6 +451,7 @@ func (bw *bufferedPageWriter) Close(hasDict, fallback bool) error {
 
 	buf := bw.inMemSink.Finish()
 	defer buf.Release()
+	bw.inMemSink.Release()
```

Comment:
This is interesting actually. Instead of adding this call to release, is it possible for us to make sure the call to `Finish()` properly releases things?

Reply:
Yeah I think that would be cleaner. The culprit is the call to `BufferWriter.Reserve(0)` to reset the buffer. It's not actually possible to reserve 0 bytes currently, the minimum capacity is 256 bytes due to the line `newCap := utils.Max(b.buffer.Cap()+b.offset, 256)`.

Setting this to simply `newCap := b.buffer.Cap()+b.offset` should fix it, but I wasn't sure if there were existing cases where the minimum buffer size was useful. We could also treat `BufferWriter.Reset(0)` as a special case and keep the buffer empty, whereas nonzero values would get rounded up to 256 since the intent is clearly to allocate some memory in that case. What do you think?

Reply:
Treating `Reset(0)` as a special case is likely the best way to go here so that we can avoid requiring the extra release

Reply:
Sounds good. I pushed up the change. I like this approach because anyone currently using `Reset` will still get a ready to use buffer, it just won't allocate any memory if zero bytes are requested.